probe-rs/src/debug/debug_info.rs

```diff
@@ -1120,6 +1121,35 @@ impl DebugInfo {
     }
 }
 
+/// This is a fuzzy 'equality' check for path names.
+/// If both paths are absolute, then it will use the standard `Path::eq` method.
+/// If the secondary path is relative, then it will succeeed if the primary ends with the compared path,
+/// e.g. `/home/user/project/src/main.rs` ends with `src/main.rs` and will return true.
+pub(crate) fn path_fuzzy_eq(primary_path: &Path, secondary_path: &Path) -> bool {
+    if primary_path.is_absolute() && secondary_path.is_relative() {
+        // First make sure the secondary path does not start with special sequences, e.g. `./` or `../`
+        let stripped_secondary_path = secondary_path
+            .components()
+            .skip_while(|component| {
+                component
+                    .as_os_str()
+                    .to_str()
+                    // If the component is not a string, then it is not a special sequence.
+                    .unwrap_or("")
+                    .starts_with('.')
```

Comment:
Does this also remove hidden directories, e.g. something like `.config`?

Reply:
Yes, it would. Not sure how often there are source files in hidden directories, but the match would continue with whatever is left. Further refinement can be an enhancement.

Reply:
See new implementation please.